test/e2e/operator/nmstate_install_test.go

```diff
@@ -12,103 +12,87 @@ import (
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	"github.com/nmstate/kubernetes-nmstate/test/cmd"
 	"github.com/nmstate/kubernetes-nmstate/test/e2e/daemonset"
 	"github.com/nmstate/kubernetes-nmstate/test/e2e/deployment"
-
-	"github.com/nmstate/kubernetes-nmstate/test/cmd"
 	testenv "github.com/nmstate/kubernetes-nmstate/test/env"
 )
 
 var _ = Describe("NMState operator", func() {
 	Context("when installed for the first time", func() {
 		BeforeEach(func() {
-			installNMState(defaultNMState)
+			By("Install NMState for the first time")
+			installNMState(defaultOperator.nmstate)
 		})
-		It("should deploy daemonset and webhook deployment", func() {
-			daemonset.GetEventually(handlerKey).Should(daemonset.BeReady())
-			deployment.GetEventually(webhookKey).Should(deployment.BeReady())
+		It("should deploy a ready operand", func() {
+			eventuallyOperandIsReady(defaultOperator)
 		})
 		AfterEach(func() {
-			uninstallNMState(defaultNMState)
+			uninstallNMStateAndWaitForDeletion(defaultOperator)
 		})
-	})
-	Context("when NMState is installed", func() {
-		It("should list one NMState CR", func() {
-			installNMState(defaultNMState)
-			daemonset.GetEventually(handlerKey).Should(daemonset.BeReady())
-			ds, err := daemonset.GetList(handlerLabels)
-			Expect(err).ToNot(HaveOccurred(), "List daemon sets in namespace nmstate succeeds")
-			Expect(ds.Items).To(HaveLen(1), "and has only one item")
-		})
-		Context("and another CR is created with a different name", func() {
-			var nmstate = defaultNMState
-			nmstate.Name = "different-name"
+		Context("and another CR is created with different name", func() {
+			var differentNMState = defaultOperator.nmstate
+			differentNMState.Name = "different-name"
 			BeforeEach(func() {
-				err := testenv.Client.Create(context.TODO(), &nmstate)
-				Expect(err).ToNot(HaveOccurred(), "NMState CR with different name is ignored")
+				eventuallyOperandIsReady(defaultOperator)
+				installNMState(differentNMState)
 			})
-			AfterEach(func() {
-				err := testenv.Client.Delete(context.TODO(), &nmstate, &client.DeleteOptions{})
-				Expect(err).ToNot(HaveOccurred(), "NMState CR with incorrect name is removed without error")
-			})
-			It("should ignore it", func() {
-				ds, err := daemonset.GetList(handlerLabels)
-				Expect(err).ToNot(HaveOccurred(), "Daemon set list is retreieved without error")
-				Expect(ds.Items).To(HaveLen(1), "and still only has one item")
+			It("should remove NMState with different name", func() {
+				Eventually(func() error {
+					return testenv.Client.Get(context.TODO(), types.NamespacedName{Name: differentNMState.Name}, &differentNMState)
+				}, 120*time.Second, 1*time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(apierrors.IsNotFound, BeTrue())))
```

Comment:
I was worrie that passing nil to IsNotFound would break something but looks like is not 

```
package main                                                                    
                                                                                
import (                                                                        
    "fmt"                                                                       
    apierrors "k8s.io/apimachinery/pkg/api/errors"                              
)                                                                               
                                                                                
func main() {                                                                   
    fmt.Println(apierrors.IsNotFound(nil))                                                                                  
}         
```
outputs:
```
[ellorent@localhost kubernetes-nmstate]$ go run isnotfound.go 
false
```

Reply:
Done